test/integration/controller/jobset_controller_test.go

```diff
@@ -545,6 +545,44 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 				},
 			},
 		}),
+		ginkgo.Entry("[failure policy] jobs are restarted individually with Recreate strategy.", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					FailurePolicy(&jobset.FailurePolicy{
+						MaxRestarts:     1,
+						RestartStrategy: jobset.Recreate,
+					})
+			},
+			steps: []*step{
+				{
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						failJobWithOptions(&jobList.Items[0], &failJobOptions{reason: ptr.To(batchv1.JobReasonPodFailurePolicy)})
+					},
+					checkJobSetCondition: testutil.JobSetActive,
+				},
+				{
+					checkJobSetState: func(js *jobset.JobSet) {
+						matchJobSetRestarts(js, 1)
+					},
+				},
+				{
+					jobSetUpdateFn: func(js *jobset.JobSet) {
+						removeForegroundDeletionFinalizers(js, testutil.NumExpectedJobs(js)-1)
+					},
+				},
+				{
+					checkJobSetState: func(js *jobset.JobSet) {
+						ginkgo.By("checking 3/4 jobs were recreated")
+						exp := map[int]int{
+							0: 1,
+							1: testutil.NumExpectedJobs(js) - 1,
+						}
+						gomega.Eventually(testutil.NumJobsByRestartAttempt, timeout, interval).WithArguments(ctx, k8sClient, js).Should(gomega.Equal(exp))
+					},
+				},
+			},
+		}),
+		// TODO(nstogner): Add test case for BlockingRecreate!!!
```

Comment:
We should add this test case for blockingRecreate before merging

Reply:
Done